src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -42,6 +42,14 @@ def self.initialize_puppet(puppet_config)
 
     Puppet::ApplicationSupport.push_application_context(master_run_mode)
 
+    # We should never end up using the ssl_context that Puppet creates in
+    # JRuby, however Puppet will load one before it calls to
+    # `Puppet::Network::HttpPool.connection` (which is what then delegates to
+    # http client). This may cause warnings if only an http (vs https)
+    # connection is requested (though we never request an http connection in
+    # JRuby from our code).
+    Puppet.push_context({ssl_context: true})
```

Comment:
This may be incorrect if users have disabled SSL routing/using the http client for HTTP based integrations. See this line: https://github.com/puppetlabs/puppet/blob/6.7.2/lib/puppet/network/http_pool.rb#L87  However, correcting it would mean altering the behavior the HttpPool in Puppet.

Reply:
@joshcooper do you know if there's a valid use case for consumers of the `HttpPool` to request HTTP (`use_ssl: false`) connections in a running server, afaict, if so we'll need to refactor the context handling linked above, or live with a warning being emitted.

Reply:
I can imagine a report processor might want to. I'm fine removing the warning in puppet if that simplifies things.